.github/workflows/main.yml

```diff
@@ -23,28 +23,45 @@ jobs:
     
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '20'
         
-      # Java and .NET are already installed on ubuntu-latest
+      # Install .NET runtime
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: 6.0.x
+      
+      # Install Java runtime (only needed if you want to run the offical HL7 Java validator)
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'microsoft'
+          java-version: '17'
         
       - name: Firely.Terminal (GitHub Actions)
-        uses: FirelyTeam/firely-terminal-pipeline@v0.3.2
+        uses: FirelyTeam/firely-terminal-pipeline@v0.4.0
         with:
           PATH_TO_CONFORMANCE_RESOURCES: Resources/fsh-generated/resources/
+          FIRELY_TERMINAL_VERSION: 3.1.0
           #PATH_TO_EXAMPLES: Examples
-          PATH_TO_QUALITY_CONTROL_RULES: qc/custom
+          #PATH_TO_QUALITY_CONTROL_RULES: qc/custom
           DOTNET_VALIDATION_ENABLED: true
           JAVA_VALIDATION_ENABLED: true
           OUTPUT_FORMAT: RAW
           JAVA_VALIDATION_OPTIONS: -allow-example-urls true
+          JAVA_VALIDATOR_VERSION: 6.0.11
           SIMPLIFIER_USERNAME: ${{ secrets.SIMPLIFIER_USERNAME }}
           SIMPLIFIER_PASSWORD: ${{ secrets.SIMPLIFIER_PASSWORD }}
           SUSHI_ENABLED: true
+          SUSHI_VERSION: 3.5.0
           SUSHI_OPTIONS: Resources/
           EXPECTED_FAILS: VALIDATION_CONFORMANCE_DOTNET VALIDATION_CONFORMANCE_JAVA VALIDATION_EXAMPLES_JAVA
 
-      - name: Add & Commit
-        uses: EndBug/add-and-commit@v7
-        with:
-          add: 'Resources/fsh-generated/resources/'
+      - name: Check for Uncommitted Changes
```

Comment:
Great progress on optimizing our workflow pipeline! Excellent effort in achieving a fully operational status—well done to everyone involved!